### PR TITLE
chore: reorder and refactor scripts/upload.ts

### PR DESCRIPTION
**Pull request by rossilor95**
- reorder code to separate _functions_ and _main_ in 2 sections (e.g., as in [ls.ts](https://github.com/teaxyz/pantry.core/blob/05c0d227146b9e99a4f63ee40839e1b537925bc0/scripts/ls.ts#L26))
- format code to uniform style
- add `bucket` as a parameter to the `get_version` and  `put ` functions to avoid the use of global variables

This is related to https://github.com/teaxyz/pantry.core/issues/32

### CODE REVIEW

File: scripts/upload.ts [hunk 1]
```diff
@@ -10,123 +10,126 @@ args:
...
-
+import { S3, S3Bucket } from "s3";
+import { pkg as pkgutils } from "utils";
+import { useCache, useFlags, useOffLicense, usePrefix } from "hooks";
+import { Package, PackageRequirement } from "types";
+import SemVer, * as semver from "semver";
+import { basename, dirname } from "deno/path/mod.ts";
+import Path from "path";
+import { set_output } from "./utils/gha.ts";
+import { sha256 } from "./bottle.ts";
```
Comment (jhheider) [inline scripts/upload.ts:21]: Our current code style doesn't end lines with semi-colons (which accounts for the excess changes). Recommend removal to minimize changeset.

**Review by jhheider — CHANGES_REQUESTED** [review]
This is going to be difficult to thoroughly review with the addition of semicolons causing most lines to show changes. Please revert that change, as that should be placed in its own stylistic PR if desired.

File: scripts/upload.ts (hunk 1 above)
Comment (rossilor95) [reply scripts/upload.ts]: Sorry @jhheider, my fault: I didn't consider that. I removed all semicolons

**Review by jhheider — APPROVED** [review]
I have no specific objection to the reorganization, and skipping capturing global `bucket` could allow for reuse, if needed. ftr, I don't believe there's necessarily an issue with global variables in a "script", compared with something larger we might call a "program", since a script is generally a single global execution context. That said, this is perfectly good practice, and means the atoms might be reusable if needed.

**Comment by mxcl** [comment]
I approve of the sentiment, but I don't see how anyone can realistically review this without a huge amount of effort or some tool to assist.

All I see is huge chunks of red and green.

**Comment by rossilor95** [comment]
I'm really sorry for that. Maybe I should have made more commits.

**Comment by mxcl** [comment]
> I'm really sorry for that. Maybe I should have made more commits.

I'm sorry, we really appreciate the contribution. Refactor PRs are *hard* because you are basically relying on test coverage, which we have none in this file.

I guess if it breaks we'll fix it. @jhheider up to you. This is your domain.

**Comment by jhheider** [comment]
Fixing is (usually) the easy part :). I'd love the exponential backoff to reduce spurious failures; if we introduce some breakage, we'll get it.

**Comment by mxcl** [comment]
k @jhheider merge when you’re ready

**Pull request merged**